### ISSUE TITLE
`DOMDocument::$preserveWhitespace` should be reported as dynamic property. Not `$preserveWhiteSpace`

### ISSUE THREAD

**Issue by sasezaki** [vimeo/psalm#8915]
Hi. I noticed that psalm would not report `DOMDocument::$preserveWhitespace` is dynamic property.
( When after I saw this commit  https://github.com/php/doc-base/commit/9965c2b16fbeba6d2451bd300d6da85260856de9 )

php-src declares as `preserveWhiteSpace` .
https://github.com/php/php-src/blob/027add9e1bf935c2d874f736105cced87fdeb226/ext/dom/php_dom.stub.php#L679

```php
<?php

$dom = new DOMDocument;
$dom->preserveWhiteSpace = false;
$dom->preserveWhitespace = false; // should be reported as dynamic property

$dom->PreserveWhitespace = false; // ERROR: UndefinedPropertyAssignment
$dom->preservewhitespace = false; // ERROR: UndefinedPropertyAssignment

// PropertyMap  `'strictErrorChecking' => 'bool',` 
$dom->strictErrorChecking = false;
$dom->strictErrorchecking = false;
```
https://psalm.dev/r/a2760ed2e5

It seems current PropertyMap is mistaken.
> 'preserveWhitespace' => 'bool',
https://github.com/vimeo/psalm/blob/5afc3282d41c382e3803a971aaff2800aa7adc3e/dictionaries/PropertyMap.php#L116

But I don't know why currently `$dom->preserveWhiteSpace` is not reported as ERROR against case sensitive...

**Comment by weirdan** [comment]
Psalm seems to think both are present: https://psalm.dev/r/3015ffd278

**Comment by weirdan** [comment]
'preserveWhite**S**pace' likely comes from reflection, while 'preserveWhite**s**pace' is from the property map.

**Comment by sasezaki** [comment]
@weirdan 

Thanks for advice! I made PR at #8918.
